toplevel/byte/topmain.ml

```diff
@@ -156,19 +156,20 @@ let prepare ppf =
       Format.fprintf ppf "Uncaught exception: %s\n" (Printexc.to_string x);
       false
 
-(* If [name] is "", then the "file" is stdin treated as a script file. *)
-let file_argument name =
+let input_argument name =
+  let filename = Toploop.filename_of_input name in
```

Comment:
this is a minor comment (I don't think it's worth changing anything now), but the patch would have been less invasive with `let input_argument input = let name = ... in ...`.